### PR TITLE
Handle promise rejections

### PR DESCRIPTION
**Pull request by maier49**
Eliminates all ` UnhandledPromiseRejectionWarning`s except one that appears to be coming from `dojo-core/Observable`'s `toPromise` method.

* [ ] There is a related issue
* [x] All contributors have signed a CLA
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] The code passes the CI tests
* [x] Unit or Functional tests are included in the PR
* [x] The PR increases or maintains the overall unit test coverage percentage
* [x] The code is ready to be merged

### CODE REVIEW

**Review by dylans — APPROVED** [review]
I think this looks fine, though it should probably get a second review in case I missed anything.

**Review by rorticus — APPROVED** [review]
👍

**Pull request merged**
